core/server/worker/src/main/java/alluxio/worker/block/AsyncBlockRemover.java

```diff
@@ -110,41 +122,31 @@ public void shutDown() {
   }
 
   private class BlockRemover implements Runnable {
-    private String mThreadName;
-
     @Override
     public void run() {
-      mThreadName = Thread.currentThread().getName();
-      long blockToBeRemoved;
+      String threadName = Thread.currentThread().getName();
       while (true) {
-        blockToBeRemoved = INVALID_BLOCK_ID;
+        Long blockToBeRemoved = null;
         try {
           blockToBeRemoved = mBlocksToRemove.take();
           mTakeCount.inc();
           mBlockWorker.removeBlock(Sessions.MASTER_COMMAND_SESSION_ID, blockToBeRemoved);
           mRemovedSuccessCount.inc();
-          LOG.debug("Block {} is removed in thread {}.", blockToBeRemoved, mThreadName);
+          LOG.debug("Block {} is removed in thread {}.", blockToBeRemoved, threadName);
         } catch (InterruptedException e) {
           Thread.currentThread().interrupt();
           // Only log warning if interrupted not due to a shutdown.
           if (!mShutdown) {
-            LOG.warn("{} got interrupted while it was cleaning block {}.", mThreadName,
+            LOG.warn("{} got interrupted while it was cleaning block {}.", threadName,
                 blockToBeRemoved);
           }
           break;
-        } catch (IOException e) {
-          LOG.warn("IOException occurred while {} was cleaning block {}, exception is {}.",
-              mThreadName, blockToBeRemoved, e.getMessage());
-        } catch (BlockDoesNotExistException e) {
-          LOG.warn("{}: block {} may be deleted already. exception is {}.",
-              mThreadName, blockToBeRemoved, e.getMessage());
-        } catch (InvalidWorkerStateException e) {
-          LOG.warn("{}: invalid block state for block {}, exception is {}.",
-              mThreadName, blockToBeRemoved, e.getMessage());
         } catch (Exception e) {
-          LOG.warn("Unexpected exception: {}.", e);
+          LOG.warn("Failed to remove block {} instructed by master. This is best-effort and "
+              + "will be tried later. threadName {}, error {}", blockToBeRemoved,
+              threadName, e.getMessage());
```

Comment:
will this still log the stack of the Exception?

Reply:
no, neither log message before this PR (which prints `e.toString()`)